### PR TITLE
GP2-279 wagtail dashboard page

### PR DESCRIPTION
**Pull request by bobmeredith**
To do (delete all that do not apply):

 - [x] Change has a jira ticket that has the correct status.
 - [x] Changelog entry added.

### CODE REVIEW

File: core/mixins.py [hunk 1]
```diff
@@ -30,4 +30,8 @@ def get_context(self, request):
 
 class AnonymousUserRequired:
     # used by core.wagtail_hooks.anonymous_user_required
-    anonymous_user_required_redirect_url = reverse_lazy('core:dashboard')
+    anonymous_user_required_redirect_url = constants.DASHBOARD_URL
+
+
+class AuthenticatedUserRequired:
+    authenticated_user_required_redirect_url = '/login/'
```
Comment (webbyfox) [inline core/mixins.py:37]: wondering shall we create constants for this as above?

File: core/templates/core/includes/routing_bar.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                    {% for route in routes  %}
+                        <article class="dashboard__routing-block w-full-m c-1-{{routes|length}}-l">
+                            {% include_block route %}
+                        </article>
+                    {% endfor %}
+                </div>
+            </div>
+        </div>
+    </div>
+{% endif %}
```
Comment (webbyfox) [inline core/templates/core/includes/routing_bar.html:25]: Newline at the end?

File: domestic/templates/domestic/domestic_dashboard.html [hunk 3]
```diff
@@ -1,14 +1,16 @@
 {% extends 'core/base.html' %}
+{% load wagtailcore_tags %}
+{% load wagtailimages_tags %}
```
Comment (webbyfox) [inline domestic/templates/domestic/domestic_dashboard.html:3]: Is it necessary to load image tags?

File: tests/unit/domestic/factories.py [hunk 4]
```diff
@@ -1,14 +1,48 @@
 import wagtail_factories
-from factory import fuzzy
-from domestic.models import DomesticHomePage
+import factory
+import factory.fuzzy
+from domestic.models import DomesticHomePage, DomesticDashboard
+# from wagtail.tests.utils.form_data import streamfield
+# from wagtail.core.fields import StreamField
+# from wagtail.core import blocks
```
Comment (webbyfox) [inline tests/unit/domestic/factories.py:7]: Remove unnecessary import, please?

**Comment by webbyfox** [comment]
@bobmeredith Some minor comments but looks good.

File: core/mixins.py (hunk 1 above)
Comment (bobmeredith) [reply core/mixins.py:37]: Yes good plan - There are a whole load of export plan literals that could go - but I didn't want to do that here.
I'll do this one sure,

File: core/templates/core/includes/routing_bar.html (hunk 2 above)
Comment (bobmeredith) [reply core/templates/core/includes/routing_bar.html:25]: Does flake8 not check HTML? I guess not.

File: tests/unit/domestic/factories.py (hunk 4 above)
Comment (bobmeredith) [reply tests/unit/domestic/factories.py:7]: Whoops  Good spot.

File: domestic/templates/domestic/domestic_dashboard.html (hunk 3 above)
Comment (bobmeredith) [reply domestic/templates/domestic/domestic_dashboard.html]: They can both go in fact - they are loaded in the include now.

**Pull request merged**